src/compiler/objective_c_generator.cc

```diff
@@ -176,65 +176,71 @@ void PrintMethodImplementations(Printer *printer,
 
 string GetHeader(const ServiceDescriptor *service, const string prefix) {
   string output;
-  grpc::protobuf::io::StringOutputStream output_stream(&output);
-  Printer printer(&output_stream, '$');
+  {
+    // Scope the output stream so it closes and finalizes output to the string.
```

Comment:
Let's make clear that we know that's what's _supposed_ to happen when the function returns anyway, but we've found it's not always the case. Or just:
```cpp
// The extra scope is needed due to gRPC issue #1769
```
Whichever way to word it you prefer, as long as it's specific so someone doesn't find this in a year and removes an apparently-unnecessary scope.